test/TritonGPU/load-mma-specialization.mlir

```diff
@@ -734,13 +734,13 @@ tt.func @matmul_tma_acc_with_conditional_def_and_use_no_multibuf_flag(
 
     // CHECK-NEXT: scf.if [[DO_EPILOGUE]]
     scf.if %do_epilogue {
-      // CHECK-NEXT: ttng.wait_barrier [[ACC_READY_BUF0]], [[ACC_PHASE]] {ttg.partition = 0 : i32}
       // CHECK-NEXT: "some_op"()
       "some_op"() : () -> ()
+      // CHECK-NEXT: ttng.wait_barrier [[ACC_READY_BUF0]], [[ACC_PHASE]] {ttg.partition = 0 : i32}
       // CHECK-NEXT: [[C:%.*]], [[USER_TOK:%.*]] = ttng.tmem_load [[ACC_BUF]][]
+      // CHECK-NEXT: ttng.arrive_barrier [[ACC_EMPTY_BUF0]], 1 {ttg.partition = 0 : i32}
```

Comment:
does membar place bar.sync here?

Reply:
yeah, I saw that `arrive_barrier` has a write effect, so I supposed that's what's happening. By inter-warp group analysis, I meant the dependency between tmem read in epilog and write in the MMA warp. In the epilog locally, the sync between tmem_load -> arrive_barrier is not necessary. So I'd say we are just lucky here. It also suggests that we might be getting other superfluous syncs like this.